hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemAuthorization.java

```diff
@@ -36,331 +40,485 @@
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_THEN_READ_ONLY_PATH).close();
-    intercept(AbfsAuthorizationException.class,
-        ()-> {
-          fs.append(TEST_WRITE_THEN_READ_ONLY_PATH).close();
+    Path testFilePath = new Path(
+        TEST_WRITE_THEN_READ_ONLY_PATH_PREFIX + name.getMethodName());
+
+    ((MockAbfsAuthorizer) fs.getAbfsStore().getAuthorizer())
+        .setwriteThenReadOnly(testFilePath.getName(), WriteReadMode.WRITE_MODE);
+    fs.create(testFilePath).close();
+
+    ((MockAbfsAuthorizer) fs.getAbfsStore().getAuthorizer())
+        .setwriteThenReadOnly(testFilePath.getName(), WriteReadMode.READ_MODE);
+    intercept(AbfsAuthorizationException.class, () -> {
+      fs.append(testFilePath).close();
     });
   }
 
   @Test
-  public void testRenameAuthorized() throws Exception {
+  public void testRenameSourceUnauthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.rename(TEST_READ_WRITE_FILE_PATH_0, TEST_READ_WRITE_FILE_PATH_1);
+    Path testSrcFilePath = new Path(
+        TEST_READ_ONLY_FILE_PATH_PREFIX_0 + name.getMethodName());
+    Path testDestFilePath = new Path(
+        TEST_READ_WRITE_FILE_PATH_PREFIX_0 + name.getMethodName());
+
+    intercept(AbfsAuthorizationException.class, () -> {
+      fs.rename(testSrcFilePath, testDestFilePath);
+    });
   }
 
   @Test
-  public void testRenameUnauthorized() throws Exception {
+  public void testRenameDestUnauthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    intercept(AbfsAuthorizationException.class,
-        ()-> {
-          fs.rename(TEST_WRITE_ONLY_FILE_PATH_0, TEST_WRITE_ONLY_FILE_PATH_1);
+    Path testSrcFilePath = new Path(
+        TEST_READ_WRITE_FILE_PATH_PREFIX_0 + name.getMethodName());
+    Path testDestFilePath = new Path(
+        TEST_READ_ONLY_FILE_PATH_PREFIX_1 + name.getMethodName());
+
+    intercept(AbfsAuthorizationException.class, () -> {
+      fs.rename(testSrcFilePath, testDestFilePath);
     });
   }
 
   @Test
   public void testDeleteFileAuthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_ONLY_FILE_PATH_0).close();
-    fs.delete(TEST_WRITE_ONLY_FILE_PATH_0, false);
+    Path testFilePath = new Path(
+        TEST_WRITE_ONLY_FILE_PATH_PREFIX_0 + name.getMethodName());
+    fs.create(testFilePath).close();
+    fs.delete(testFilePath, false);
   }
 
   @Test
   public void testDeleteFileUnauthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_THEN_READ_ONLY_PATH).close();
-    intercept(AbfsAuthorizationException.class,
-        ()-> {
-          fs.delete(TEST_WRITE_THEN_READ_ONLY_PATH, false);
+    Path testFilePath = new Path(
+        TEST_WRITE_THEN_READ_ONLY_PATH_PREFIX + name.getMethodName());
+
+    ((MockAbfsAuthorizer) fs.getAbfsStore().getAuthorizer())
```

Comment:
pull this out into `getMockAuthorizer() {return getFileSystem().getAbfsStore.getAuthorizer()}`

Reply:
Refactored